vendor/bat-native-ads/src/bat/ads/internal/notifications.cc

```diff
@@ -49,76 +49,91 @@ void Notifications::Initialize(InitializeCallback callback) {
 bool Notifications::Get(const std::string& id, NotificationInfo* info) const {
   DCHECK(is_initialized_);
 
-  auto notification = notifications_.find(id);
-  if (notification == notifications_.end()) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  *info = notification->second;
+  *info = *iter;
 
   return true;
 }
 
-void Notifications::Add(const NotificationInfo& info) {
+void Notifications::PushBack(const NotificationInfo& info) {
   DCHECK(is_initialized_);
 
-  notifications_.insert({info.id, info});
+  notifications_.push_back(info);
+  ads_client_->ShowNotification(std::make_unique<NotificationInfo>(info));
 
   SaveState();
 }
 
-bool Notifications::Remove(const std::string& id) {
+void Notifications::PopFront(bool should_dismiss) {
+  if (!notifications_.empty()) {
+    if (should_dismiss){
+      ads_client_->CloseNotification(notifications_.front().id);
+    }
+    notifications_.pop_front();
+    SaveState();
+  }
+}
+
+bool Notifications::Remove(const std::string& id, bool should_dismiss) {
   DCHECK(is_initialized_);
 
-  if (!Exists(id)) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  ads_client_->CloseNotification(id);
-  notifications_.erase(id);
+  if (should_dismiss){
+    ads_client_->CloseNotification(id);
+  }
+  notifications_.erase(iter);
 
   SaveState();
 
   return true;
 }
 
-void Notifications::RemoveAll() {
+void Notifications::RemoveAll(bool should_dismiss) {
   DCHECK(is_initialized_);
 
-  for (const auto& notification : notifications_) {
-    auto id = notification.first;
-    Remove(id);
+  if (should_dismiss){
+    for (const auto& notification : notifications_) {
+      ads_client_->CloseNotification(notification.id);
+    }
   }
+  notifications_.clear();
 
   SaveState();
 }
 
-void Notifications::CloseAll() const {
-  DCHECK(is_initialized_);
-
-  for (const auto& notification : notifications_) {
-    auto id = notification.first;
-    ads_client_->CloseNotification(id);
-  }
-}
-
 bool Notifications::Exists(const std::string& id) const {
   DCHECK(is_initialized_);
 
-  if (notifications_.find(id) == notifications_.end()) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
   return true;
 }
 
+uint64_t Notifications::Count() const {
```

Comment:
Please change to `int`

Reply:
uint64_t mimics deque::size_type, which is unsigned by default. uint64_t is used to satisfy both 32 and 64 bit targets.

Reply:
https://google.github.io/styleguide/cppguide.html#Integer_Types `We use int very often, for integers we know are not going to be too big, e.g., loop counters. Use plain old int for such things.`

Reply:
It's not about this case. 
I don't think it's correct to generate warnings converting unsigned to signed or 64 to 32 bit types (when on arm64).

Reply:
Our size will never exceed UINT_MAX and does not rely on 32-bit modular arithmetic. however as this is a nit I am resolving issue as it is only a return type and we are not persisting these 64 bit numbers.